packages/web-app-files/src/composables/actions/files/useFileActionsCreateNewFile.ts

```diff
@@ -0,0 +1,265 @@
+      return
+    }
+
+    try {
+      const path = join(unref(currentFolder).path, fileName)
+      const resource = await (clientService.webdav as WebDAV).putFileContents(unref(currentSpace), {
+        path
+      })
+
+      if (loadIndicatorsForNewFile.value) {
+        resource.indicators = getIndicators({ resource, ancestorMetaData: unref(ancestorMetaData) })
+      }
+
+      store.commit('Files/UPSERT_RESOURCE', resource)
+
+      if (openAction) {
+        openEditor(
+          openAction,
+          unref(currentSpace).getDriveAliasAndItem(resource),
+          resource.webDavPath,
+          resource.fileId,
+          EDITOR_MODE_CREATE
+        )
+        store.dispatch('hideModal')
+
+        return
+      }
+
+      store.dispatch('hideModal')
+      store.dispatch('showMessage', {
+        title: $gettext('"%{fileName}" was created successfully', { fileName })
+      })
+    } catch (error) {
+      console.error(error)
+      store.dispatch('showMessage', {
+        title: $gettext('Failed to create file'),
+        status: 'danger'
+      })
+    }
+  }
+
+  const handler = (
+    fileActionOptions: FileActionOptions,
+    extension: string,
+    addAppProviderFile: boolean,
+    openAction: any // FIXME: type?
+  ) => {
+    const checkInputValue = (value) => {
+      store.dispatch(
+        'setModalInputErrorMessage',
+        checkNewFileName(areFileExtensionsShown.value ? value : `${value}.${extension}`)
+      )
+    }
+    let defaultName = $gettext('New file') + `.${extension}`
+
+    if (unref(files).some((f) => f.name === defaultName)) {
+      defaultName = resolveFileNameDuplicate(defaultName, extension, unref(files))
+    }
+
+    if (!areFileExtensionsShown.value) {
+      defaultName = extractNameWithoutExtension({ name: defaultName, extension } as any)
+    }
+
+    const inputSelectionRange = !areFileExtensionsShown.value
+      ? null
+      : [0, defaultName.length - (extension.length + 1)]
+
+    const modal = {
+      variation: 'passive',
+      title: $gettext('Create a new file'),
+      cancelText: $gettext('Cancel'),
+      confirmText: $gettext('Create'),
+      hasInput: true,
+      inputValue: defaultName,
+      inputLabel: $gettext('File name'),
+      inputError: checkNewFileName(
+        areFileExtensionsShown.value ? defaultName : `${defaultName}.${extension}`
+      ),
+      inputSelectionRange,
+      onCancel: () => store.dispatch('hideModal'),
+      onConfirm: addAppProviderFile
+        ? addAppProviderFileFunc
+        : (fileName) => {
+            if (!areFileExtensionsShown.value) {
+              fileName = `${fileName}.${extension}`
+            }
+            addNewFile(fileName, openAction)
+          },
+      onInput: checkInputValue
+    }
+
+    store.dispatch('createModal', modal)
+  }
+
+  const actions = computed((): FileAction[] => {
+    const actions = []
+    for (const newFileHandler of newFileHandlers || []) {
+      const addAppProviderFile = false
```

Comment:
I believe you can even remove `addAppProviderFile` and just pass `openAction`. In `handler` where we check `addAppProviderFile ? ...` we can simply check `!openAction ? ...`